### PR TITLE
Support version 3.4.2

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- added version 3.4.2
- removed version 3.4.1

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [x] (Integration-)Test cases added
- [x] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by maltesander** [comment]
test https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/spark-k8s-operator-it-custom/126/

File: tests/test-definition.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ dimensions:
       - "false"
   - name: spark
     values:
-      - 3.4.1
```
Comment (siegfriedweber) [inline tests/test-definition.yaml:8]: 3.4.1 is deprecated but still supported and therefore should be tested.

File: docs/modules/spark-k8s/partials/supported-versions.adoc [hunk 2]
```diff
@@ -3,6 +3,7 @@
 // Stackable Platform documentation.
 // Please sort the versions in descending order (newest first)
 
-- 3.5.0 (Hadoop 3.3.4, Scala 2.12, Python 3.11, Java 11)
-- 3.4.1 (Hadoop 3.3.4, Scala 2.12, Python 3.11, Java 11)
-- 3.4.0 (Hadoop 3.3.4, Scala 2.12, Python 3.11, Java 11) (deprecated)
+- 3.5.0 (Hadoop 3.3.4, Scala 2.13, Python 3.11, Java 11)
+- 3.4.2 (Hadoop 3.3.4, Scala 2.13, Python 3.11, Java 11) (LTS)
+- 3.4.1 (Hadoop 3.3.4, Scala 2.13, Python 3.11, Java 11) (deprecated)
```
Comment (siegfriedweber) [inline docs/modules/spark-k8s/partials/supported-versions.adoc:8]: Where does the Scala version come from?

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: docs/modules/spark-k8s/partials/supported-versions.adoc (hunk 2 above)
Comment (maltesander) [reply docs/modules/spark-k8s/partials/supported-versions.adoc]: i had it confused with kafka...spark is 2.12...reverted https://github.com/stackabletech/spark-k8s-operator/pull/357/commits/7b995835d063eaf5195715b902e3ee995a82b23c

File: tests/test-definition.yaml (hunk 1 above)
Comment (maltesander) [reply tests/test-definition.yaml:8]: readded 3.4.1, fixed example and ny jars https://github.com/stackabletech/spark-k8s-operator/pull/357/commits/83f1421c709ac70eb57382f01695a5b03d9a4bf3

**Review by siegfriedweber — APPROVED** [review]
LGTM

**Pull request merged**